internal/connection_manager.go

```diff
@@ -191,16 +193,10 @@ func (cm *connectionManagerImpl) getConnection(address core.Address, asOwner boo
 	if !found {
 		return nil
 	}
-
-	if !asOwner {
-		return conn
-	}
-
-	if conn.isOwnerConnection {
-		return conn
+	if cm.shouldAuthenticateConn(asOwner, conn) {
```

Comment:
The method name does not correctly explain the intention here. getConnection does not authenticate.
This is more like a check to see if the connection meets the owner requirement if asked. 
I would prefer not to extract function here. The code is not really complicated, and will probably explain itself better. 
Same for the other usage of this method. There, the connection is already authenticated. It is to see if it is authenticated as the owner if asked that way. Again, I would prefer the leave it as is.

Reply:
So this check is to see if we are trying to get connection as owner, then the connection is indeed the owner connection right?

In the other usage, if the check is true then we authenticate the connection. If it is already authenticated then why do we authenticate it once more?

The same code is used twice in different methods and the checks are the same checks, so I would prefer to have a method for that.